openedx/adg/lms/applications/managers.py

```diff
@@ -122,6 +122,42 @@ def accepted_student_count(self, business_lines):
             status=application_models.UserApplication.ACCEPTED, business_line__in=business_lines
         ).count()
 
+    def filter_by_query_params(
+        self, name_or_email=None, omni_course_statuses=None, bu_course_statuses=None, business_unit_titles=None
+    ):
+        """
+        Filter the applications by the given query params
+
+        Arguments:
+            name_or_email (str): Name/Email to filter queryset by using user profile's name or email
+            omni_course_statuses (list): Statuses to filter the queryset's program course status
+            bu_course_statuses (list): Statuses to filter the queryset's bu course statuses
+            business_unit_titles (list): Business unit titles to filter the queryset
+
+        Returns:
+            QuerySet: Filtered queryset
+        """
+        applications = self
+        if name_or_email:
+            applications = applications.filter(
+                Q(user__profile__name__icontains=name_or_email) | Q(user__email=name_or_email)
```

Comment:
In django admin, it searches for partial email address as well, like if I provide the search string "arbisoft", it will fetch all users belonging to the arbisoft organization in the User table (by filtering all users that have 'arbisoft' in their email addresses.

Reply:
This was the way it was being done in the performance dashboard. We can do an `icontains` search here too but it is an expensive operation

Reply:
Alright let's go with it then. I just thought that an admin may not remember the complete and exact email address of the person they want to search, or they may want to search all applications of users from a particular organization, etc.